static/src/js/components/PermissionForm/PermissionForm.jsx

```diff
@@ -0,0 +1,647 @@
+import Form from '@rjsf/core'
+import React, { useEffect, useState } from 'react'
+
+import Button from 'react-bootstrap/Button'
+import Col from 'react-bootstrap/Col'
+import Container from 'react-bootstrap/Container'
+import Row from 'react-bootstrap/Row'
+
+import validator from '@rjsf/validator-ajv8'
+import collectionPermission from '@/js/schemas/collectionPermission'
+
+import collectionPermissionUiSchema from '@/js/schemas/uiSchemas/collectionPermission'
+
+import saveTypesToHumanizedStringMap from '@/js/constants/saveTypesToHumanizedStringMap'
+
+import saveTypes from '@/js/constants/saveTypes'
+
+import { useNavigate, useParams } from 'react-router'
+import { useMutation, useSuspenseQuery } from '@apollo/client'
+
+import useAppContext from '@/js/hooks/useAppContext'
+import useNotificationsContext from '@/js/hooks/useNotificationsContext'
+
+import errorLogger from '@/js/utils/errorLogger'
+import removeEmpty from '@/js/utils/removeEmpty'
+
+import { CREATE_ACL } from '@/js/operations/mutations/createAcl'
+import { UPDATE_ACL } from '@/js/operations/mutations/updateAcl'
+
+import CollectionSelectorPage from '@/js/pages/CollectionSelectorPage/CollectionSelectorPage'
+import { isEmpty } from 'lodash-es'
+import {
+  GET_COLLECTION_FOR_PERMISSION_FORM
+} from '@/js/operations/queries/getCollectionForPermissionForm'
+import CustomArrayFieldTemplate from '../CustomArrayFieldTemplate/CustomArrayFieldTemplate'
+import CustomDateTimeWidget from '../CustomDateTimeWidget/CustomDateTimeWidget'
+import CustomFieldTemplate from '../CustomFieldTemplate/CustomFieldTemplate'
+import CustomSelectWidget from '../CustomSelectWidget/CustomSelectWidget'
+import CustomTextWidget from '../CustomTextWidget/CustomTextWidget'
+import CustomTitleField from '../CustomTitleField/CustomTitleField'
+import CustomTitleFieldTemplate from '../CustomTitleFieldTemplate/CustomTitleFieldTemplate'
+import GridLayout from '../GridLayout/GridLayout'
+import GroupPermissionSelect from '../GroupPermissionSelect/GroupPermissionSelect'
+import KeywordPicker from '../KeywordPicker/KeywordPicker'
+
+import ChooseProviderModal from '../ChooseProviderModal/ChooseProviderModal'
+
+/**
+ * Validates the form data for the access constraints and temporal constraints.
+ *
+ * @param {Object} formData - The data submitted in the form.
+ * @param {Object} errors - The errors object to record validation errors.
+ * @returns {Object} - The errors object with any validation errors added.
+ */
+const validate = (formData, errors) => {
+  const collectionMinValue = formData.accessConstraintFilter
```

Comment:
Destructure these values, avoid chains of multiple calls into nested objects and avoid repeated code.

It is difficult to follow this logic, adding comments on each check about what it is actually validating might be useful.

I think this function should also be pulled out into a util and unit tested there